addons/gamification/models/gamification_challenge.py

```diff
@@ -614,84 +566,47 @@ def report_progress(self, cr, uid, challenge, context=None, users=False, subset_
 
-            ctx.update({'challenge_lines': lines_boards})
-            body_html = temp_obj.render_template(cr, uid, challenge.report_template_id.body_html, 'gamification.challenge', challenge.id, context=ctx)
+            body_html = Temp.with_context(challenge_lines=lines_boards).render_template(self.report_template_id.body_html, 'gamification.challenge', self.id)
 
             # send to every follower and participant of the challenge
-            self.message_post(cr, uid, challenge.id,
+            self.message_post(
                 body=body_html,
-                partner_ids=[user.partner_id.id for user in challenge.user_ids],
-                context=context,
+                partner_ids=[user.partner_id.id for user in self.user_ids],
                 subtype='mail.mt_comment')
-            if challenge.report_message_group_id:
-                self.pool.get('mail.channel').message_post(cr, uid, challenge.report_message_group_id.id,
+            if self.report_message_group_id:
+                self.report_message_group_id.message_post(
                     body=body_html,
-                    context=context,
                     subtype='mail.mt_comment')
 
         else:
             # generate individual reports
-            for user in users or challenge.user_ids:
-                goals = self._get_serialized_challenge_lines(cr, uid, challenge, user.id, restrict_goal_ids=subset_goal_ids, context=context)
+            for user in users or self.user_ids:
+                goals = self._get_serialized_challenge_lines(user.id, restrict_goal_ids=subset_goal_ids)
                 if not goals:
                     continue
 
-                ctx.update({'challenge_lines': goals})
-                body_html = temp_obj.render_template(cr, user.id,  challenge.report_template_id.body_html, 'gamification.challenge', challenge.id, context=ctx)
+                body_html = Temp.with_context(challenge_lines=goals).render_template(self.report_template_id.body_html, 'gamification.challenge', self.id)
 
                 # send message only to users, not on the challenge
-                self.message_post(cr, uid, 0,
-                                  body=body_html,
-                                  partner_ids=[(4, user.partner_id.id)],
-                                  context=context,
-                                  subtype='mail.mt_comment')
-                if challenge.report_message_group_id:
-                    self.pool.get('mail.channel').message_post(cr, uid, challenge.report_message_group_id.id,
-                                                             body=body_html,
-                                                             context=context,
-                                                             subtype='mail.mt_comment')
-        return self.write(cr, uid, challenge.id, {'last_report_date': fields.date.today()}, context=context)
+                self._model.message_post(
```

Comment:
instead of this 
you can try to use new(values={}) method which return new record instance and
        initialized with the provided ``value``. This new record is *not* created
        in database, it only exists in memory.
then call message_post